app/controllers/api/v1/dqt_records_controller.rb

```diff
@@ -14,6 +14,12 @@ def show
           head :not_found
         end
       end
+
+    private
+
+      def access_scope
+        ApiToken.where(private_api_access: true)
+      end
```

Comment:
I don't know which level of granularity we want really. Is it enough for our use case to have these permissions on the level of STI?

So in here, you could do something like `NpqRegistrationApiToken.all`?

Do we expect some tokens of particular type to have more access than other tokens of particular type? Like, having a live E&L token that can access DQT, and test E&L tokens which can't? Likewise, do we ever expect to give some LP tokens the right to use this endpoint, but not all?

Reply:
> So in here, you could do something like NpqRegistrationApiToken.all?

We can do that but then you've coupled the entire controller to one type of token. No other token can be used. Whereas the current implementation doesn't care what token it gets as long as the flag is true.

 > Do we expect some tokens of particular type to have more access than other tokens of particular type? Like, having a live E&L token that can access DQT, and test E&L tokens which can't? Likewise, do we ever expect to give some LP tokens the right to use this endpoint, but not all?

With the mechanism I have suggested you'd be able to set certain tokens access to private apis if you so wished. This gives us full flexibility. If we scope to `NpqRegistrationApiToken.all` you've dug yourself a hole.

Reply:
> With the mechanism I have suggested you'd be able to set certain tokens access to private apis if you so wished.

Yeah, you would be able to. Would we want to? 

Are we confident that there is an existing use case for it? If not, should we postpone adding the functionality until there is?

Reply:
> Yeah, you would be able to. Would we want to?

The answer to that is I don't know. I think none one does. But if we needed to, we can now do that easily.

> Are we confident that there is an existing use case for it? If not, should we postpone adding the functionality until there is?

The suggested change currently meets our need to be able to make certain api endpoints private and for those private apis only be accessible to certain tokens. I don't see any benefit here of postponing.

Reply:
Fine, I am worried that we will eventually get some concrete requirements that conflict with this setup, but then again, removing a boolean is pretty trivial.